### ISSUE TITLE
Add test coverage for error events

### ISSUE THREAD

**Issue by PrasadG193** [kubeshop/botkube#340]
**Describe the bug**

In e2e test suite, we mock resource creation with fake client and verify the BotKube event generations.
https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go#L46

We need to add similar test cases to verify error event handling.
We can use client-go [Event interface](https://godoc.org/k8s.io/client-go/kubernetes/typed/core/v1#EventInterface) for creating new event

Resource configuration can be mocked by setting/altering values in `utils.AllowedUpdateEventsMap` and `utils.AllowedEventKindsMap` https://github.com/infracloudio/botkube/blob/develop/pkg/utils/utils.go#L148

Add test cases to validate the following behavior:

- [ ] Validate error events are handled for a resource with `events: [all]` in test config https://github.com/infracloudio/botkube/blob/develop/test/resource_config.yaml - @Divya063

- [ ] Validate error events are handled with only `events: [error]` config for a resource - @Divya063

- [ ]  Validate error events are skipped when `error` event is not configured for a resource in test config. e.g `events: [create, update, delete]` - @Divya063

- [x] Validate that error events are skipped when an event occurs for a namespace which is not configured

- [x] Validate that error events are skipped when an event occurs for a resource which is not added in resource_config - @Divya063

Each test case can be added as a separate PR. Multiple PR as expected for this task. 
Guidelines for contributors:
- Post comment on the issue informing others about the test case you want to add
- Mark the issue as done once the PR is merged
- Please make sure that no one else is working on the same test case to avoid duplicate PRs

**Comment by sheetalsingala** [comment]
Please can I work on this?

**Comment by PrasadG193** [comment]
Go for it @sheetalsingala. Please let me know if you have any doubts.

**Comment by PrasadG193** [comment]
Hey @sheetalsingala I've updated the issue description about the test cases we should add. Feel free to choose one or more from the list

**Comment by Divya063** [comment]
@PrasadG193 I would like to work on the following test cases:
- Validate error events are handled for a resource with events: [all] in test config
- Validate error events are handled with only events: [error] config for a resource
- Validate error events are skipped when error event is not configured for a resource in test config. e.g events: [create, update, delete]
- Validate that error events are skipped when an event occurs for a resource which is not added in resource_config

**Comment by PrasadG193** [comment]
@Divya063 awesome! Please let me know if you need any help

**Comment by archit041198** [comment]
Hi @PrasadG193 I would like to work on Validate that error events are skipped when an event occurs for a namespace which is not configured

**Comment by pkosiec** [comment]
Hi, 
I'm closing this task as not valid anymore, as we switched to the E2E tests with real Slack and Kubernetes APIs and we have covered the feature in E2E and unit tests.
We can discuss new ideas under this epic: https://github.com/kubeshop/botkube/issues/589.

Thanks everyone involved!
